js/models/base.js

```diff
@@ -515,4 +515,16 @@ export class PageList extends List {
         this.populate();
         return this;
     }
+
+    /**
+     * Perform a server-side search
+     * @param  {String} query The query string to perform the search on.
+     * @return {Object} Return itself allowing to chain methods.
+     */
+    search_server(query) {
```

Comment:
I prefer:
```javascript
on_fetched(data) {
    super.on_fetched(data);
    this.page = 1;  // Clear the paging
}
```
And then simply call `fetch({q: 'my seach'})`.
Like that it does not create a special path.